### ISSUE TITLE
2.4.11 Focus Visible (Enhanced) Outline Offset Success Criterion

### ISSUE THREAD

**Issue by shrthnd** [w3c/wcag#1056]
Hi all, a colleague and I were reviewing the new success criterion for enhanced, visible focus and were wondering about how this edge-case would be handled: 

If the "focus indication area" is greater than or equal to the longest side of a bounding rectangle (multiplied by 0.0213) but the bounding rectangle sits at the top or bottom of the screen and spans the entire width, only a single edge of the focus ring will be visible. This is more or less a problem depending on how many sides of an element bump up against the edge of a screen.

An alternative approach would be to use a negative outline-focus which will pull the focus-ring inside of the bounding rectangle.

<img width="667" alt="Screen Shot 2020-02-28 at 2 31 05 PM" src="https://user-images.githubusercontent.com/25191/75581204-1168bf80-5a37-11ea-93b8-8ab213838c42.png">

Am I misreading the new criterion, or is there a reason this is less desirable than a single outline appearing along the long-edge of an element?

Thanks in advance for any feedback and let me know if I can clarify my question or use-case.

**Comment by JAWS-test** [comment]
I think it is about the visible focus, not the theoretical focus. This means that if the focus is not visible because it is outside the viewport, for example, only the visible focus is included in the calculation. This can mean that the criterion is violated if its size is insufficient

**Comment by alastc** [comment]
Hi @shrthnd,

Thanks for commenting! 

> If the "focus indication area" is greater than or equal to the longest side of a bounding rectangle (multiplied by 0.0213)

Where did you gather 0.0213 from? I'm guessing that's the angle of the CSS pixel, but I recommend sticking to 2 CSS pixels for our conversations :-) 

I agree it is (or can be) a problem that an outline or other indicator is cut off by the viewport, but that is not something the author can necessarily control. It would vary by screensize, wrapping of text, and other factors. If it gets cut off by the content, e.g disappears behind another block, that would be the author's issue though and could lead to a fail.

Using an inner focus does help with both scenarios (content & viewport), and an example of that is included in the materials. I didn't think it was worth the complexity of including that though, do you? (If so, I'd love to hear some ideas on how to include that!)

**Comment by dbseagull** [comment]
Interesting stuff. Three of us on a team responsible for implementing the Guidelines on new sites and retrofitting old sites have been obsessing over this. Perhaps we're oversimplifying, but what if something like this were said?

The focus indication area is 2 CSS pixels greater than or less than the longest side of the bounding rectangle.

**Comment by shrthnd** [comment]
> Where did you gather 0.0213 from?

Good catch, @alastc. I probably meant 0.0426 (2 * visual angle of about 0.0213 degrees) but 2 CSS pixels is certainly easier to understand and doesn't require any of these complex maths.

> The focus indication area is 2 CSS pixels greater than or less than the longest side of the bounding rectangle.

This makes sense to me conceptually, @dbseagull, when we're talking about the area contained _within_ the focus indication area, often indicated (in web browsers) as a focus-outline/ring pseudo element--by default it is equivalent to the "bounding rectangle" of a control or element.

This was the root of my confusion here. In retrospect, the focus indication area is not the area indicated as "in focus"--it is the indicator itself: "the number of CSS pixels that change between the focused and unfocused states of a User Interface Component"

> Minimum area: The focus indication area is greater than or equal to the longest side of the bounding rectangle of the focused control, times 2 CSS pixels.

@JAWS-test you make a good point and I appreciate this insight: 

> ...it is about the visible focus, not the theoretical focus. This means that if the focus is not visible because it is outside the viewport, for example, only the visible focus is included in the calculation. This can mean that the criterion is violated if its size is insufficient.

Revisiting the definition:

> Minimum area: The focus indication area [the number of CSS pixels that change between the focused and unfocused states of a User Interface Component] is greater than or equal to the longest side of the bounding rectangle of the focused control, times 2 CSS pixels.

The "focus indication area" is calculated by the longest-edge of the "bounding rectangle" (multiplied by 2 CSS px) in this definition. 

Does this mean a qualifying focus indication area must be a bounding rectangle? 

My gut says no: a bounding rectangle is only one way to indicate focus, which must be 2px thick with sufficient contrast, and include X pixels (calculated by multiplying the longest edge of the visible element/control by 2 PX.) Any additional indication of focus/hover would require an equivalent minimum number of pixels to change in order to be used as a success criterion.

At the end of the day, the definition of how focus-indication area is calculated (based on the bounding rectangle) made me think the focus-indication area should/could only ever be a bounding rectangle, which has a whole mess of other implications. This doesn't appear true on closer inspection so I'm closing the issue. 

Thank you all for the feedback.

**Comment by alastc** [comment]
> The "focus indication area" is calculated by the longest-edge of the "bounding rectangle" (multiplied by 2 CSS px) in this definition.
> Does this mean a qualifying focus indication area must be a bounding rectangle?

No, I think that might be the source of the confusion.

The focus indicator area is the _change_, which could be a bounding box, or a background change, or an icon, or anything really. The complexity is coming from the flexibility.

Thanks for closing it, but I will take this aspect as a potential update to the text.
